lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,455 @@
+
+        Args:
+            cube (iris.cube.Cube):
+                Radius will be applied equally on x and y dimensions.
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube of same shape as cube with haloes applied.
+        """
+        new_cube = self.neighbourhood.process(cube.copy())
+        return new_cube
+
+    def _update_metadata(self, cube):
+        """
+        Modify the meta data of input cube to resemble a Nowcast of lightning
+        probability.
+        1. Rename to "lightning_probability"
+        2. Remove "threshold" coord
+
+        Args:
+            cube (iris.cube.Cube):
+                An input cube
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube - a copy of input cube with meta-data relating to
+                a Nowcast of lightning probability.
+                The data array will be a copy of the input cube.data
+        """
+        new_cube = cube.copy()
+        new_cube.rename("lightning_probability")
+        new_cube.remove_coord('threshold')
+        if self.debug:
+            print(('In {}, new_cube is {}'.format(self, new_cube)))
+        return new_cube
+
+    def _modify_first_guess(self, cube, first_guess_lightning_cube,
+                            prob_lightning_cube, prob_precip_cube,
+                            prob_vii_cube):
+        """
+        Modify first-guess lightning probability with nowcast data
+
+        Args:
+            cube (iris.cube.Cube):
+                Provides the meta-data for the Nowcast lightning probability
+                output cube
+
+            first_guess_lightning_cube (iris.cube.Cube):
+                First-guess lightning probability
+                Must have same x & y dimensions as cube
+                Time dimension should overlap that of cube
+
+            prob_lightning_cube (iris.cube.Cube):
+                Nowcast lightning rate
+                Must have same dimensions as cube
+
+            prob_precip_cube (iris.cube.Cube):
+                Nowcast precipitation probability (threshold > 0.5, 7, 35)
+                Must have same other dimensions as cube
+
+            prob_vii_cube (iris.cube.Cube):
+                Radar-derived vertically integrated ice content (VII)
+                Must have same x and y dimensions as cube
+                Time should be a scalar coordinate
+                Must have a threshold coordinate with points matching
+                self.vii_thresholds
+                Can be <No cube> or None or anything that evaluates to False
+
+        Returns:
+            new_cube (iris.cube.Cube):
+                Output cube containing Nowcast lightning probability.
+        """
+        new_cube_list = iris.cube.CubeList([])
+        for cube_slice in cube.slices_over('time'):
+            this_point = cube_slice.coord('time').points[0]
+            this_time = iris_time_to_datetime(
+                cube_slice.coord('time').copy())[0]
```

Comment:
Almost inclined to suggest updating "iris_time_to_datetime" so that it doesn't modify the coord in place.  Would halve the amount of code you have in this loop!

Reply:
I decided not to!